app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -4,7 +4,7 @@ module Users
   class OmniauthCallbacksController < Devise::OmniauthCallbacksController
     def login_dot_gov
       @kind = 'Login.gov'
-      @email = auth_hash['info']['email'] if auth_hash && auth_hash['info']['email_verified']
+      @email = auth_hash['info']['all_emails'].find { |email| email.end_with(".gov") || email.end_with(".mil") } if auth_hash && auth_hash['info']['email_verified']
```

Comment:
select the first gov or mil email, rather than using the `email` which is the last email used to login